lib/config/presets/index.spec.ts

```diff
@@ -346,6 +346,128 @@ describe('config/presets/index', () => {
         ],
       });
     });
+
+    it('resolves and keeps extends array from config ', async () => {
+      config.extends = ['local>username/preset-repo'];
+      local.getPreset.mockResolvedValueOnce({
+        labels: ['self-hosted resolved'],
+      });
+      const res = await presets.resolveConfigPresets(
+        config,
+        {},
+        [],
+        [],
+        true // shallow log config
+      );
+      expect(res).toEqual({
+        labels: ['self-hosted resolved'],
+      });
+    });
+
+    it('resolves shallow config in case of local and github extend', async () => {
+      config.extends = ['config:base', 'some/repo'];
+      config.packageRules = [
+        {
+          matchManagers: ['github-actions'],
+          groupName: 'github-actions dependencies',
+        },
+      ];
+      local.getPreset.mockResolvedValueOnce({
+        extends: ['github>username/preset-repo'],
+        labels: ['self-hosted resolved'],
+      });
+      gitHub.getPreset.mockResolvedValueOnce({
+        packageRules: [
+          {
+            matchDatasources: ['docker'],
+            matchPackageNames: ['ubi-minimal'],
+            versioning: 'regex',
+          },
+        ],
+      });
+      const res = await presets.resolveConfigPresets(
+        config,
+        {},
+        [],
+        [],
+        true // shallow log config
+      );
+      expect(res).toEqual({
+        packageRules: [
+          {
+            matchDatasources: ['docker'],
+            matchPackageNames: ['ubi-minimal'],
+            versioning: 'regex',
+          },
+          {
+            matchManagers: ['github-actions'],
+            groupName: 'github-actions dependencies',
+          },
+        ],
+        extends: ['config:base'],
+        labels: ['self-hosted resolved'],
+      });
+    });
+
+    it('combines extends with user and github', async () => {
+      config.extends = ['config:base', 'github>username/preset-repo'];
+      config.packageRules = [
+        {
+          matchManagers: ['github-actions'],
+          groupName: 'github-actions dependencies',
+        },
+      ];
+      gitHub.getPreset.mockResolvedValueOnce({
+        extends: ['config:js-app'],
+      });
+      const res = await presets.resolveConfigPresets(
+        config,
+        {},
+        [],
+        [],
+        true // shallow log config
+      );
+      expect(res).toEqual({
+        extends: ['config:base', 'config:js-app'],
+        packageRules: [
+          {
+            matchManagers: ['github-actions'],
+            groupName: 'github-actions dependencies',
+          },
+        ],
+      });
+    });
+
+    it('ignore fetching internal persets ', async () => {
+      // ignore resolving of merge-confidence:beta
+      config.extends = [
+        'config:base',
+        'github>whitesource/merge-confidence:beta',
```

Comment:
should we have a test for `whitesource/merge-confidence:beta` ? as it's a shorthand `local>whitesource/merge-confidence:beta`